### DISCUSSION TITLE
REopt Web Tool Timeout Error Message

### DISCUSSION THREAD

**Discussion by tetsu0228** [NREL/REopt-API-Analysis#85] [1 - General REopt Questions] [answered]
Hi, I get an error message with ID: b8c7e8ab-2520-4eb8-ac80-c6b9bb3a04e8 Since yesterday.

**Comment by LParkhill** [comment, marked as answer]
There are quick steps that you can take to reduce the optimization complexity and give it a better chance of completing within the time limit. 
 
* The first thing to try is to limit the range of data points that the optimization must consider. To do this, limit maximum capacity inputs. 
   ** PV and battery default maximum capacity inputs are “unlimited”. You can enter a lower maximum, perhaps 5000 kW, and save the optimization the extra time of looking at the possibility of sizing all the way to the default maximum of “unlimited”. If your results hit 5000 kW, you can always raise the maximum for a subsequent run. 
 
* If those changes don’t make enough impact for the evaluation to complete, the next input that can cause timeouts is complex utility rates. Sometimes it’s possible to create a custom rate from your utility rate that has been simplified a bit by removing cost tiers – if this is an acceptable change to make in your evaluation.
 
* The final option for scenarios that are too complex to solve within the time limit is the “tolerance” input in the Site and Utility section of the webtool. The default for the input is 0.1% (or 0.001), but you can increase it up to 5%: 
   ** The solver optimality tolerance is the threshold for the difference between the solution’s objective value (life cycle cost) and the best possible value (lower bound of the objective function as determined by the optimization model) at which the solver terminates. Note, there is no guarantee that the best possible value would be achieved if the model ran longer. It’s possible that the solution achieved within the optimality tolerance is the same solution that would be found if the model ran indefinitely.
   ** Try increasing this value to 2-3% if no solution is found within the model’s timeout limit. Increase the value further if the model still times out. The maximum allowed tolerance value is 5%. Once a solution is found with the higher tolerance, you can choose to bound the technology sizes using the minimum and maximum size inputs and run the model with a lower tolerance.
 
Please give these a try. If they don’t work, let us know and we can look at other solutions.

**Reply by tetsu0228** [reply]
Thanks. It went well!!
